File: 卡码/代码随想录/1/08-贪心算法/Java/greedy/src/main/java/com/kurumi/greedy/_02_中等/LeetCode_0135.java
//135. 分发糖果
package com.kurumi.greedy._02_中等;

import java.util.Arrays;

public class LeetCode_0135 {

}
class Solution {
  public int candy(int[] ratings) {
    int[] can=new int[ratings.length];
    Arrays.fill(can, 1);
    for(int i=1;i<ratings.length;i++)
      if(ratings[i]>ratings[i-1]) can[i]=can[i-1]+1;
    for(int i=ratings.length-2;i>=0;i--)
      if(ratings[i]>ratings[i+1]) can[i]=Integer.max(can[i+1]+1,can[i]);
    int sum=0;
    for(int val:can) sum+=val;
    return sum;
  }
}

File: 卡码/代码随想录/C++/贪心/02-中等/LeetCode_0135.cpp
//135. ·Ö·¢ÌÇ¹û
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

class Solution {
public:
  int candy(vector<int>& ratings) {
		vector<int> can(ratings.size(),1);
		for(int i=1;i<ratings.size();i++)
			if(ratings[i]>ratings[i-1]) can[i]=can[i-1]+1;
		for(int i=ratings.size()-2;i>=0;i--)
			if(ratings[i]>ratings[i+1]) can[i]=max(can[i+1]+1,can[i]);
		return accumulate(can.begin(),can.end(),0);
  }
};

int main(){
  system("pause");
  return 0;
}
